### PR TITLE
Fix/cleanup create invoice service

### PR DESCRIPTION
**Pull request by da-kami**
This cleans up the service instantiation and will (eventually) allow us to spin up the UI with purely mocked services :)

@klochowicz were working on this, he will drive this forward and clean up `rust` calls in `main`. :)

### CODE REVIEW

File: mobile/lib/features/wallet/send_screen.dart [hunk 1]
```diff
@@ -28,26 +28,25 @@ class SendScreen extends StatefulWidget {
...
+    initChannelValues(channelInfoService);
+    final WalletService walletService = context.read<WalletChangeNotifier>().service;
+    invoiceFromClipboard(walletService);
     super.initState();
   }
 
-  Future<void> initChannelValues() async {
+  Future<void> initChannelValues(ChannelInfoService channelInfoService) async {
     channelInfo = await channelInfoService.getChannelInfo();
   }
```
Comment (klochowicz) [inline mobile/lib/features/wallet/send_screen.dart:47]: it might be easier in fact to not have this method, as line 46 gives more / better information 🤔

**Review by klochowicz — APPROVED** [review]
thanks for the cleanup!

File: mobile/lib/features/wallet/send_screen.dart (hunk 1 above)
Comment (da-kami) [reply mobile/lib/features/wallet/send_screen.dart:47]: Adding this method does not change any behavor but it makes me more aware that I am calling something `async`. This is more a stylistic pattern I'm using - it makes me more aware that I am calling `async` code from the `initState` method. As far as I know we cannot `await` in `initState`.
If I just put the call to the underlying service in there I won't see that the call is `async`. If you feel that's not useful feel free to remove it - I'll leave it for now.

**Pull request merged**